### PR TITLE
[Overflow-117] [FE] Integrate api for answer form and component

### PR DESCRIPTION
**Pull request by presyos**
## Backlog Link
https://framgiaph.backlog.com/view/SUN_OVERFLOW-117
## Commands
None
## Pre-conditions
None
## Expected Output
Can add answer
## Notes
Route: http://localhost:3000/questions/1
Added the Zustand and Toaster in this PR
The placement of the answer component is not yet fixed because there are still components above the answer form so this is for the meantime only
## Screenshots
image
image
image
image

### CODE REVIEW

File: frontend/components/organisms/AnswerComponent/index.tsx [hunk 1]
```diff
@@ -1,27 +1,80 @@
...
+            user_id: data.question_id,
+            question_id: data.user_id
+          }
+        })
+
+        toast.promise(new_answer, {
+          loading: 'Adding Answer...',
+          success: 'Answer Successfully Added!',
+          error: 'There was an Error!',
+        });
```
Comment (kenorenzsun) [inline frontend/components/organisms/AnswerComponent/index.tsx:46]: You can add a plugin file where you have an export function for success, error, info toast

Comment (kenorenzsun) [reply frontend/components/organisms/AnswerComponent/index.tsx:46]:
```suggestion
// example plugin/helper file 
import toast, { Toaster } from 'react-hot-toast';

export function formProcessToast(promise, loadingMessage = "Processing ...", successMessage = "Process done", errorMessage = "Error in process",position = "top-right") {
    return toast.promise(promise, {
        loading: loadingMessage,
        success: successMessage,
        error: errorMessage,
    });
}
```

Noticed that it does not have position prop maybe you can try [React toastify](https://www.npmjs.com/package/react-toastify)

File: frontend/helpers/graphql/mutations/create_answer.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { gql } from "@apollo/client";
+
+const CREATE_ANSWER = gql`
+    mutation CreateAnswer( 
+    $content: String!
+    $user_id: ID! 
+    $question_id: ID!){
```
Comment (kenorenzsun) [inline frontend/helpers/graphql/mutations/create_answer.tsx:7]:
```suggestion
    mutation CreateAnswer( 
        $content: String!
        $user_id: ID! 
        $question_id: ID!
    ){
```

File: frontend/helpers/graphql/mutations/create_answer.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        createAnswer(
+            content: $content 
+            user_id: $user_id 
+            question_id: $question_id){
```
Comment (kenorenzsun) [inline frontend/helpers/graphql/mutations/create_answer.tsx:11]:
```suggestion
            question_id: $question_id
        ){
```

File: frontend/components/organisms/AnswerComponent/index.tsx [hunk 4]
```diff
@@ -1,27 +1,80 @@
...
-            type="submit"
-            isDisabled={false}
-            onClick={handleOnClick}
-          >
-            Submit Answer
-          </Button>
-        </div>
+        <form
+          className="flex flex-col"
+        >
```
Comment (nicoleamber) [inline frontend/components/organisms/AnswerComponent/index.tsx:60]: Format properly

`<form className="flex flex-col">`

**Review by nicoleamber — APPROVED** [review]

File: frontend/helpers/store.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { create } from 'zustand'
+interface UserState {
+    user_id: number
+    addUserID: (user: number) => void
+}
+
+const useUsertore = create<UserState>()((set) => ({
```
Comment (jules-lucero) [inline frontend/helpers/store.tsx]: rename to `useUserStore`

**Review by jules-lucero — APPROVED** [review]

**Comment by ipilikent** [comment]
## Issues 
- The user's input answer does not disappear when the answer is submitted, it disappears when it is submitted


## screenshot/GIF

**Comment by ipilikent** [comment]
## Approved PR
-  The user can submit answer
- The error message is displayed

## Proof
<details>

117 -FE Answer Component


</details>

Test Case: https://docs.google.com/spreadsheets/d/114GrEn-LQQ8saIvjz9zqenuk05qmO4dN/edit#gid=1031094368 
-----
- Total Test time: 1 hr.
- Total Comments: 1
- Total Iterations: 2

**Pull request merged**